sdk/daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/UpgradeCheckMain.scala

```diff
@@ -26,13 +27,20 @@ case class UpgradeCheckMain(loggerFactory: NamedLoggerFactory) {
   implicit val ec: ExecutionContext = ExecutionContext.global
   implicit val loggingContext: LoggingContextWithTrace = LoggingContextWithTrace.empty
 
-  private def decodeDar(
+  private def decodeDarOrDalf(
       path: String
-  ): Either[CouldNotReadDar, Dar[(Ref.PackageId, Ast.Package)]] = {
+  ): Either[CouldNotReadDarOrDalf, Dar[(Ref.PackageId, Ast.Package)]] = {
     val s: String = s"Decoding DAR from ${path}"
     logger.debug(s)
     val result = DarDecoder.readArchiveFromFile(new File(path))
```

Comment:
side note, it is recommenced to prefer `java.nio.*` over `java.io.*`, (`Path` are better than `File`).

Reply:
I only use File in order to construct a File to pass to readArchiveFromFile - is there a variant that takes a Path? I think we can leave this to a separate PR anyways for now, because I've reverted all of my changes to UpgradeCheckMain.scala